test/core_checkM_test.py

```diff
@@ -228,32 +490,26 @@ def test_checkM_lineage_wf_full_app_single_assembly(self):
         print ("=================================================================\n")
 
         # run checkM lineage_wf app on a single assembly
-        input_ref = self.assembly_ref1
+        assembly = TEST_DATA['assembly_list'][3]
+
+        input_ref = getattr(self, assembly['attr'])
         params = {
+            'dir_name': assembly['attr'],
             'workspace_name': self.ws_info[1],
             'input_ref': input_ref,
             'reduced_tree': 0,
-            # 'save_output_dir': 0,  # DEBUG
-            'save_output_dir': 1,  # DEBUG
+            'save_output_dir': 1,
             'save_plots_dir': 1,
             'threads': 4
         }
-        result = self.getImpl().run_checkM_lineage_wf(self.getContext(), params)[0]
-
-        pprint('End to end test result:')
-        pprint(result)
-
-        self.assertIn('report_name', result)
-        self.assertIn('report_ref', result)
-
-        # make sure the report was created and includes the HTML report and download links
-        rep = self.getWsClient().get_objects2({'objects':
-                                              [{'ref': result['report_ref']}]})['data'][0]['data']
-
-        self.assertEquals(rep['direct_html_link_index'], 0)
-        self.assertEquals(len(rep['file_links']), 3)
-        self.assertEquals(len(rep['html_links']), 1)
-        self.assertEquals(rep['html_links'][0]['name'], 'CheckM_Plot.html')
+        expected_results = {
```

Comment:
Refactored all the end to end checkM tests to use the `run_and_check_report` method, since they are all in the same format.